demo/authenticator.py

```diff
@@ -0,0 +1,105 @@
+import mesop as me
+
+@me.stateclass
+class State:
+    username: str = ""
+    password: str = ""
+    welcome_message: str = ""
+    error_message: str = ""
+
+def on_blur_username(e: me.InputBlurEvent):
+    state = me.state(State)
+    state.username = e.value
+
+def on_blur_password(e: me.InputBlurEvent):
+    state = me.state(State)
+    state.password = e.value
+
+def on_login(e: me.ClickEvent):
+    state = me.state(State)
+    
+    # Validate username and password
+    if not state.username or not state.password:
+        state.error_message = "Please enter both username and password."
+        return
+    
+    # Simulate a simple login check (replace with actual authentication logic)
+    if state.username and state.password:
+        state.welcome_message = f"Welcome, {state.username}!"
+        state.error_message = ""  # Clear any previous error message
+        # Navigate to the welcome page after successful login
+        me.navigate("/welcome")
+    else:
+        state.error_message = "Incorrect username or password."
+    
+    # Clear the username and password fields
+    state.username = ""
```

Comment:
Good idea to clear these fields, especially password. 

Sorry for the delay on this review. I've just been thinking about if it's safe in Mesop to store the password in state. Ideally, we'd only send the password for the log in check (but never send it back). But I don't think that's currently possible in Mesop. The blur event helps in that at least we're not repeatedly sending parts of the password back and forth via the on input event.

I'm still not sure at the moment.